lxd/storage/drivers/driver_ceph_volumes.go

```diff
@@ -644,7 +644,165 @@ func (d *ceph) CreateVolumeFromMigration(vol VolumeCopy, conn io.ReadWriteCloser
 
 // RefreshVolume updates an existing volume to match the state of another.
 func (d *ceph) RefreshVolume(vol VolumeCopy, srcVol VolumeCopy, refreshSnapshots []string, allowInconsistent bool, op *operations.Operation) error {
-	return genericVFSCopyVolume(d, nil, vol, srcVol, refreshSnapshots, true, allowInconsistent, op)
+	var lastCommonSnapshotName string
+	lastCommonSnapshotIndex, found := d.findLastCommonSnapshot(vol.Snapshots, refreshSnapshots)
+	if found {
+		_, lastCommonSnapshotName, _ = api.GetParentAndSnapshotName(vol.Snapshots[lastCommonSnapshotIndex].name)
+	}
+
+	if lastCommonSnapshotName != "" {
+		// Remove all snapshots from the target that will get refreshed.
+		// Those are all the snapshots after the last common snapshot.
+		lastCommonSnapshotFound := false
+		for _, targetSnapshot := range vol.Snapshots {
```

Comment:
Rather than going through the snapshot list again and doing name matching, why not use the index returned from d.findLastCommonSnapshot to start iterating the vol.Snapshots from the next snapshot onward (assuming there is one).

Reply:
Yes, I very much like this approach, with this we can also get rid of the two if conditions here.

Reply:
Ok I tried starting to iterate from the the `lastCommonSnapshot`. In case there aren't any snapshots yet on the target, we need extra logic to instead start at the first snapshot.
Additionally with the current way of doing it we can easily access the predecessor snapshot using `i-1` which we need to create the delta between two snaps. We could just start at last `lastCommonSnapshot-1` and skip the first loop but then this would again require logic if there isn't yet a last common snapshot. 
Tbh I am convinced this logic is the simplest out of all with the least amount of exceptions. Maybe I am overseeing something.